### DISCUSSION TITLE
Why should I use Kiali Operator?

### DISCUSSION THREAD

**Discussion by barddes** [kiali/kiali#7521] [Q&A]
### What do you want to discuss?

What is the difference between running the Operator vs the Server (helm charts)? From what I've seen, the operator uses the provided Kiali CR to install a Kiali Server helm release in my cluster.

The documentation says that the operator is the recommended way for deploying [(1)](https://kiali.io/docs/installation/installation-guide/) and that the documented features may not be available without the Operator deployment option [(2)](https://kiali.io/docs/installation/installation-guide/install-with-helm/).

Why shouldn't I just install the kiali-server chart directly? What features am I missing from running it without the operator?

### What does it improve / What problem does it solve?

To clarify the difference between methods for installing Kiali, allowing users to decide which install works the best for its own use case.

### What is the recommended proposal?

Update the docs with the differences between both install methods

### What are the alternate solutions?

**Comment by jmazzitelli** [comment]
There isn't anything documented for this specifically yet, but off the top of head, `deployment.accessible_namespaces` allows for regex expressions - the operator performs the regex and replaces that field with the full list of matching namespaces at the time of install. The server helm chart does not. The server helm chart only supports cluster-wide access - so that means people usually leave accessible namespaces to be the value `["**"]` when using server helm chart installs.

The operator can be configured to watch the cluster's namespaces - as more namespaces are created, the operator automatically forces a re-reconciliation of the Kiali CR - if that namespace needs to be added to the list of namespaces accessible to Kiali, the operator will do so. Helm chart obviously doesn't do this.

Another thing I think the operator does that the server helm chart does not do is process the different secret auth fields ([some auth fields accept values like `secret:secret_name:secret_key`](https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.external_services)). The operator will mount those secrets to the kiali server deployment pod. I'm not 100% sure, but I think the server helm chart does not support that feature.

Finally, one major reason for the existence of the operator in the first place is because operators are a first-class installation concept in OpenShift, so to plug Kiali seamlessly into an OpenShift cluster ecosystem, having an operator is how to do that.
